Fix hod_module tests and move cmd line verification to cmd line verification function.

ehiggs opened the pull request.

Commit:
Add --hod-module to create and batch tests.

Commit:
Move check for `--hod-module` from hodjob to hod.option

Since that's where cmd line parting is done.

boegel commented on hod/options.py:
```diff
@@ -64,4 +64,8 @@ def validate_pbs_option(options):
     if not options.workdir:
         _log.error('No workdir ("--workdir") provided')
         return False
+    if not options.hod_module:
```
This function is also called for `genconfig`, does it make sense to require `--hod-module` for `genconfig`?

ehiggs replied:
> does it make sense to require --hod-module for genconfig?

Not really. Fixed in a subsequent commit.

Commit:
Split out genconfig's validation into another function.

There is some copy paste here, but trying to remove the copy paste is difficult
since it's more than just 'make sure all these params are there' due to the xor
on hodconf and dist.

boegel commented on hod/options.py:
```diff
@@ -64,4 +64,22 @@ def validate_pbs_option(options):
+
+    return True
+
+def validate_genconfig_option(options):
```
rename to `validate_required_options` and call this from `validate_pbs_options`?

avoid duplicate code, it'll bite us in the ass later if we don't...

Commit:
Derp derp derp derp. I am an idiot. Thanks @boegel.
